### PR TITLE
Feat: Add database model persistence

### PR DESCRIPTION
**Pull request by aanorbel**

### CODE REVIEW

File: app/src/main/java/org/openobservatory/ooniprobe/common/AppDatabase.java [hunk 1]
```diff
@@ -11,7 +11,7 @@
 @Database(name = AppDatabase.NAME, version = AppDatabase.VERSION, foreignKeyConstraintsEnforced = true)
 public class AppDatabase {
     public static final String NAME = "v2";
-    public static final int VERSION = 3;
+    public static final int VERSION = 3 + 2;
```
Comment (bassosimone) [inline app/src/main/java/org/openobservatory/ooniprobe/common/AppDatabase.java:14]: ?

File: app/src/main/java/org/openobservatory/ooniprobe/model/database/Result.java [hunk 2]
```diff
@@ -216,10 +232,28 @@ public void delete(Context c) {
...
+             * If the descriptor does not exist, then this is an OONI Provided test or an OONI Run v1 measurement result.
+             * We return an {@link OONIDescriptor} object which implements {@link AbstractDescriptor}.
+             */
             return Optional.of(OONITests.valueOf(test_group_name.toUpperCase()).toOONIDescriptor(context));
         } catch (IllegalArgumentException e) {
+            /**
+             * If there is an {@link IllegalArgumentException}
+             * This should only happen when the test_group_name is not a valid {@link OONITests} value,
+             * Which means the `test_group_name` is not an OONI provided test or an installed `Descriptor`.
+             * Orphan resulta for an uninstalled OONI Run v2 descriptor would fall into this category and thus should not exist.
```
Comment (bassosimone) [inline app/src/main/java/org/openobservatory/ooniprobe/model/database/Result.java:254]:
```suggestion
             * Orphan results for an uninstalled OONI Run v2 descriptor would fall into this category and thus should not exist.
```

File: app/src/main/java/org/openobservatory/ooniprobe/model/database/TestDescriptor.kt [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+    override fun isEnabled(preferenceManager: PreferenceManager): Boolean {
+        return !testDescriptor.isArchived
+    }
+
+    override fun getRuntime(context: Context, preferenceManager: PreferenceManager): Int {
+        return R.string.TestResults_NotAvailable
+    }
+
+
```
Comment (bassosimone) [inline app/src/main/java/org/openobservatory/ooniprobe/model/database/TestDescriptor.kt:98]:
```suggestion
```

**Review by bassosimone — APPROVED** [review]
🐳

File: app/src/main/java/org/openobservatory/ooniprobe/common/AppDatabase.java (hunk 1 above)
Comment (aanorbel) [reply app/src/main/java/org/openobservatory/ooniprobe/common/AppDatabase.java:14]: I am adding `+2` here so i don't forget why the increment is 2. first increment is for modifications done to the schema ,`Migration4` and another for the addition of a new model `TestDescriptor`

**Pull request merged**